data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/ProcessWorker.java

```diff
@@ -50,7 +50,7 @@ public ProcessWorker(
         this.pipeline = pipeline;
         this.pluginMetrics = PluginMetrics.fromNames("ProcessWorker", pipeline.getName());
         this.invalidEventHandlesCounter = pluginMetrics.counter(INVALID_EVENT_HANDLES);
-        this.acknowledgementsEnabled = pipeline.getSource().areAcknowledgementsEnabled();
+        this.acknowledgementsEnabled = pipeline.getSource().areAcknowledgementsEnabled() || readBuffer.isByteBuffer();
```

Comment:
I'm not sure this logic is right. A byte buffer in itself does not require acknowledgements. Can we add a new default method on `Buffer` to express the actual need?

```
boolean isPersistentBuffer();
```

Or some other boolean?